core/src/main/java/nu/yona/server/goals/entities/ITimezoneGoal.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2020 Stichting Yona Foundation
+ * This Source Code Form is subject to the terms of the Mozilla Public License, v.2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package nu.yona.server.goals.entities;
+
+import nu.yona.server.analysis.entities.DayActivity;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+
+public interface ITimezoneGoal extends IGoal
+{
+	private int[] determineSpreadOutsideGoal(DayActivity dayActivity)
+	{
+		int[] spread = dayActivity.getSpread().stream().mapToInt(Integer::intValue).toArray();
+		getSpreadCellsIntStream().forEach(i -> spread[i] = 0);
+		return spread;
+	}
+
+	default int computeTotalMinutesBeyondGoal(DayActivity dayActivity)
+	{
+		int[] spread = determineSpreadOutsideGoal(dayActivity);
+		int sumOfSpreadOutsideGoal = Arrays.stream(spread).sum();
+		// Due to rounding, the sum of the spread might be more than the total duration, so take the lowest of the two
+		return Math.min(dayActivity.getTotalActivityDurationMinutes(), sumOfSpreadOutsideGoal);
+	}
+
+	default boolean isGoalAccomplished(DayActivity dayActivity)
+	{
+		int[] spread = determineSpreadOutsideGoal(dayActivity);
+		return !Arrays.stream(spread).anyMatch(i -> (i > 0));
+	}
+
+
+	default public IntStream getSpreadCellsIntStream()
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Reorder the modifiers to comply with the Java Language Specification. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AModifiersOrderCheck)